File: src/elixir/svd/lib/svd/models/fresh_water.ex
defmodule SVD.Models.FreshWater do
  @moduledoc false

  @fields [
    :fresh_water_produced,
    :fresh_water_consumed,
    :fresh_water_received,
    :fresh_water_discharged,
    :fresh_water_rob,
    :fresh_water_bunkered,
    :technical_water_produced,
    :technical_water_consumed,
    :wash_water_consumed,
    :fresh_water_remaining
  ]

  @derive {Jason.Encoder, only: @fields}
  defstruct @fields

  @type t :: %__MODULE__{}

  @spec fields() :: [atom()]
  def fields, do: @fields
end

File: src/go/svd/models/fresh_water.go
package models

// FreshWater represents the "Fresh Water" section of SVD.
type FreshWater struct {
	FreshWaterBunkered     float64 `json:"freshWaterBunkered" xml:"FreshWaterBunkered" csv:"FreshWaterBunkered"`
	FreshWaterProduced     float64 `json:"freshWaterProduced" xml:"FreshWaterProduced" csv:"FreshWaterProduced"`
	FreshWaterConsumed     float64 `json:"freshWaterConsumed" xml:"FreshWaterConsumed" csv:"FreshWaterConsumed"`
	TechnicalWaterProduced float64 `json:"technicalWaterProduced" xml:"TechnicalWaterProduced" csv:"TechnicalWaterProduced"`
	TechnicalWaterConsumed float64 `json:"technicalWaterConsumed" xml:"TechnicalWaterConsumed" csv:"TechnicalWaterConsumed"`
	WashWaterConsumed      float64 `json:"washWaterConsumed" xml:"WashWaterConsumed" csv:"WashWaterConsumed"`
	FreshWaterRemaining    float64 `json:"freshWaterRemaining" xml:"FreshWaterRemaining" csv:"FreshWaterRemaining"`
}
